rir/src/compiler/translations/rir_2_pir/rir_2_pir_compiler.cpp

```diff
@@ -29,7 +30,14 @@ Rir2PirCompiler::Rir2PirCompiler(Module* module) : RirCompiler(module) {
 }
 
 Closure* Rir2PirCompiler::compileClosure(SEXP closure) {
-    assert(isValidClosureSEXP(closure));
+    assert(TYPEOF(closure) == CLOSXP);
+    if (!isValidClosureSEXP(closure)) {
+        Protect p;
```

Comment:
The `Protect` is not needed here, no? Neither `BODY` nor `SET_BODY` trigger GC (I hope), and it is destroyed immediately after.